### PR TITLE
Use v1.0.0-dev branch as new master

### PR DESCRIPTION
**Pull request by chrismayer**
This PR will set the development branch [v1.0.0-dev](https://github.com/chrismayer/wegue/tree/v1.0.0-dev) as the new master. A lot of basic / foundation work and improvements were done in this branch in order to improve app development with Wegue.
So Wegue will tackle some things now in an easier way and the separation of application code and core code is improved. Thus upgrading will be easier since all app-specific code is now managed in a separate folder `app`.

A quick overview what has been done:

  - Handle all app-specific code in a separate folder
  - Introducing an app template which can be used in the app folder
  - Script to init the app folder correctly 
  - Overhauled unit test mechanism
  - Increased test coverage
  - Vector styling improvements (from JSON config)
  - Fixes and cleanups
  - ...

See the single commits for details. 

The current master is preserved in the [v0.9.x-maintenance](https://github.com/meggsimum/wegue/tree/v0.9.x-maintenance) branch and will offer legacy code for projects based on that code. But the main focus of development will shift to the "new" master introduced by this PR.

I'd be happy for reviews even if I am aware that such a big PR is hard to read / review.  TIA!

### CODE REVIEW

**Comment by justb4** [comment]
I see Travis build failure, `vuetify` dep missing? maybe add `npm run init:app` in Travis build?

**Comment by justb4** [comment]
Yes, with 63 files changed quite a PR! Will just review some small details here and there and let you go ahead, within a some mins or so.

File: README.md [hunk 1]
```diff
@@ -24,33 +24,73 @@ Ensure that you have clean commits (and messages) and a meaningful description i
...
+  - Install the JS dependencies:
 
 ``` bash
 # install dependencies
 npm install
+```
 
-# serve with hot reload at localhost:8080
-npm run dev
+  - Create a file `app/WguApp.vue` acting as a base application for your project with the following content:
```
Comment (justb4) [inline README.md]: may add `npm run init:app` here ?

**Comment by justb4** [comment]
I say: go merge. My VueJS knowledge is also too basic to grasp it all ;-).

**Comment by chrismayer** [comment]
Thanks for the review @justb4, I'll merge now

**Pull request merged**